qualtran/_infra/registers.py

```diff
@@ -162,7 +162,7 @@ def build_from_dtypes(cls, **registers: QDType) -> 'Signature':
             registers: keyword arguments mapping register name to QDType. All registers
                 will be 0-dimensional and THRU.
         """
-        return cls(Register(name=k, dtype=v) for k, v in registers.items() if v.num_qubits)
+        return cls(Register(name=k, dtype=v) for k, v in registers.items())
```

Comment:
Okay, I get the motivation but I think we should pull out this change in a separate PR since it has potential to break existing stuff and add thorough tests.

Reply:
Is the scope of the tests above zero-sized registers specifically, or more generally for the behavior of the `Unitary` and `TensorProduct` bloqs?

(In other words, if we were to back out the zero-sized stuff, are they still needed? I responded below to your other comment about Cirq-style tests.)

Reply:
The tests were specifically for zero-sized registers.

Reply:
I have removed the zero-sized registers from this PR.